bin/elastishadow/main.c

```diff
@@ -0,0 +1,1261 @@
+        subscription_topic_filter,
+        AWS_ARRAY_SIZE(subscription_topic_filter),
+        "$aws/things/" PRInSTR "/shadow/name/" PRInSTR "/get/+",
+        AWS_BYTE_CURSOR_PRI(thing_name_cursor),
+        AWS_BYTE_CURSOR_PRI(shadow_name_cursor));
+    struct aws_byte_cursor subscription_topic_filter_cursor = aws_byte_cursor_from_c_str(subscription_topic_filter);
+
+    char accepted_path[128];
+    snprintf(
+        accepted_path,
+        AWS_ARRAY_SIZE(accepted_path),
+        "$aws/things/" PRInSTR "/shadow/name/" PRInSTR "/get/accepted",
+        AWS_BYTE_CURSOR_PRI(thing_name_cursor),
+        AWS_BYTE_CURSOR_PRI(shadow_name_cursor));
+
+    char rejected_path[128];
+    snprintf(
+        rejected_path,
+        AWS_ARRAY_SIZE(rejected_path),
+        "$aws/things/" PRInSTR "/shadow/name/" PRInSTR "/get/rejected",
+        AWS_BYTE_CURSOR_PRI(thing_name_cursor),
+        AWS_BYTE_CURSOR_PRI(shadow_name_cursor));
+
+    struct aws_byte_cursor correlation_token_path = aws_byte_cursor_from_c_str("clientToken");
+    struct aws_mqtt_request_operation_response_path response_paths[] = {
+        {
+            .topic = aws_byte_cursor_from_c_str(accepted_path),
+            .correlation_token_json_path = correlation_token_path,
+        },
+        {
+            .topic = aws_byte_cursor_from_c_str(rejected_path),
+            .correlation_token_json_path = correlation_token_path,
+        },
+    };
+
+    char publish_topic[128];
+    snprintf(
+        publish_topic,
+        AWS_ARRAY_SIZE(publish_topic),
+        "$aws/things/" PRInSTR "/shadow/name/" PRInSTR "/get",
+        AWS_BYTE_CURSOR_PRI(thing_name_cursor),
+        AWS_BYTE_CURSOR_PRI(shadow_name_cursor));
+
+    char correlation_token[128];
+    s_write_correlation_token_string(aws_byte_cursor_from_array(correlation_token, AWS_ARRAY_SIZE(correlation_token)));
+
+    char request[256];
+    snprintf(request, AWS_ARRAY_SIZE(request), "{\"clientToken\":\"%s\"}", correlation_token);
+
+    struct aws_mqtt_request_operation_options get_options = {
+        .subscription_topic_filters = &subscription_topic_filter_cursor,
+        .subscription_topic_filter_count = 1,
+        .response_paths = response_paths,
+        .response_path_count = 2,
+        .publish_topic = aws_byte_cursor_from_c_str(publish_topic),
+        .serialized_request = aws_byte_cursor_from_c_str(request),
+        .correlation_token = aws_byte_cursor_from_c_str(correlation_token),
+        .completion_callback = s_on_get_shadow_complete,
+        .user_data = aws_string_new_from_c_str(allocator, correlation_token),
+    };
+
+    printf(
+        "Submitting GetNamedShadow request for shadow '" PRInSTR "' of thing '" PRInSTR
+        "' using correlation token %s...\n",
+        AWS_BYTE_CURSOR_PRI(shadow_name_cursor),
+        AWS_BYTE_CURSOR_PRI(thing_name_cursor),
+        correlation_token);
+
+    if (aws_mqtt_request_response_client_submit_request(context->rr_client, &get_options) == AWS_OP_ERR) {
```

Comment:
Trivial: We should `aws_string_destroy` the correlation_token here if the completion isn't going to be called.

Reply:
If applicable here, applicable in all places where we check the result of `aws_mqtt_request_response_client_submit_request`